### PR TITLE
diagnostic: warn on outdated OS X versions.

### PR DESCRIPTION
**Pull request by mikemcquaid**
We don't have CI or new bottles for them so they aren't supported well so we should warn users.

### CODE REVIEW

File: Library/Homebrew/diagnostic.rb [hunk 1]
```diff
@@ -304,11 +304,19 @@ def check_for_broken_symlinks
...
+          <<-EOS.undent
+            You are using OS X #{MacOS.version}.
+            We do not provide support for this pre-release version.
+            You may encounter build failures or other breakages.
+          EOS
+        elsif OS::Mac.version < :mavericks
+          <<-EOS.undent
+            You are using OS X #{MacOS.version}.
+            We (and Apple) do not provide support for this old version.
+            You may encounter build failures or other breakages
```
Comment (bfontaine) [inline Library/Homebrew/diagnostic.rb]: There’s a missing period at the end here.

File: Library/Homebrew/diagnostic.rb [hunk 2]
```diff
@@ -304,11 +304,19 @@ def check_for_broken_symlinks
...
-        You may encounter build failures or other breakages.
-        EOS
+        return if ARGV.homebrew_developer?
+        if OS::Mac.prerelease?
+          <<-EOS.undent
+            You are using OS X #{MacOS.version}.
+            We do not provide support for this pre-release version.
+            You may encounter build failures or other breakages.
+          EOS
+        elsif OS::Mac.version < :mavericks
```
Comment (xu-cheng) [inline Library/Homebrew/diagnostic.rb]: How about adding a new method (e.g. `OS::Mac#outdated_release?`) similar to `OS::Mac#prerelease?`

Comment (UniqMartin) [inline Library/Homebrew/diagnostic.rb]: As the author of `OS::Mac#prerelease?` obviously a :+1: from me.

**Comment by UniqMartin** [comment]
Can we make this two independent checks? I realize they have both in common that they deal with an unsupported OS X release, but that's about all the overlap there is (both condition and message differ). My suggestion would be to name them `check_for_prerelease_osx` and `check_for_outdated_osx`.

**Comment by mikemcquaid** [comment]
> Can we make this two independent checks? I realize they have both in common that they deal with an unsupported OS X release, but that's about all the overlap there is (both condition and message differ).

My reasoning for a single check is that they are mutually exclusive.

**Comment by UniqMartin** [comment]
> My reasoning for a single check is that they are mutually exclusive.

That's a valid argument. Mine is that keeping them separate makes the checks more succinct and allows for the early-return-else-message pattern that I proposed elsewhere. (But I'll leave the decision to you.)

File: Library/Homebrew/diagnostic.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Homebrew/diagnostic.rb]: Added in c50de67.

**Comment by mikemcquaid** [comment]
> That's a valid argument. Mine is that keeping them separate makes the checks more succinct and allows for the early-return-else-message pattern that I proposed elsewhere. (But I'll leave the decision to you.)

Thanks!

**Comment by UniqMartin** [comment]
> Thanks!

I was obviously hoping my argument would be sufficiently persuasive, but oh well …

**Pull request closed without merging**